lib/ansible/module_utils/eos.py

```diff
@@ -17,78 +17,219 @@
-    transport=dict(default='cli', choices=['cli', 'eapi']),
-    use_ssl=dict(default=True, type='bool'),
-    provider=dict(type='dict')
-)
-
-CLI_PROMPTS_RE = [
-    re.compile(r"[\r\n]?[\w+\-\.:\/\[\]]+(?:\([^\)]+\)){,3}(?:>|#) ?$"),
-    re.compile(r"\[\w+\@[\w\-\.]+(?: [^\]])\] ?[>#\$] ?$")
-]
-
-CLI_ERRORS_RE = [
-    re.compile(r"% ?Error"),
-    re.compile(r"^% \w+", re.M),
-    re.compile(r"% ?Bad secret"),
-    re.compile(r"invalid input", re.I),
-    re.compile(r"(?:incomplete|ambiguous) command", re.I),
-    re.compile(r"connection timed out", re.I),
-    re.compile(r"[^\r\n]+ not found", re.I),
-    re.compile(r"'[^']' +returned error code: ?\d+"),
-    re.compile(r"[^\r\n]\/bin\/(?:ba)?sh")
-]
-
-
-def to_list(val):
-    if isinstance(val, (list, tuple)):
-        return list(val)
-    elif val is not None:
-        return [val]
-    else:
-        return list()
-
-
-class Eapi(object):
-
-    def __init__(self, module):
-        self.module = module
+EAPI_FORMATS = ['json', 'text']
 
-        # sets the module_utils/urls.py req parameters
-        self.module.params['url_username'] = module.params['username']
-        self.module.params['url_password'] = module.params['password']
+add_argument('use_ssl', dict(default=True, type='bool'))
+add_argument('validate_certs', dict(default=True, type='bool'))
+
+ModuleStub = collections.namedtuple('ModuleStub', 'params fail_json')
+
+def argument_spec():
+    return dict(
+        # config options
+        running_config=dict(aliases=['config']),
+        config_session=dict(default='ansible_session'),
+        save_config=dict(default=False, aliases=['save'])
+    )
+eos_argument_spec = argument_spec()
+
+def get_config(module):
+    config = module.params['running_config']
+    if not config:
+        config = module.config.get_config(include_defaults=True)
+    return NetworkConfig(indent=3, contents=config)
+
+def load_config(module, candidate):
+    config = get_config(module)
+
+    commands = candidate.difference(config)
+    commands = [str(c).strip() for c in commands]
+
+    session = module.params['config_session']
+    save_config = module.params['save_config']
+
+    result = dict(changed=False)
+
+    if commands:
+        if module._diff:
+            diff = module.config.load_config(commands, session_name=session)
+            if diff:
+                result['diff'] = dict(prepared=diff)
+
+            if not module.check_mode:
+                module.config.commit_config(session)
+                if save_config:
+                    module.config.save_config()
+            else:
+                module.config.abort_config(session_name=session)
+
+        else:
+            module.config(commands)
+            if save_config:
+                module.config.save_config()
+
+        result['changed'] = True
+        result['updates'] = commands
+
+    return result
+
+def expand_intf_range(interfaces):
+    match = INTF_NAME_RE.match(interfaces)
```

Comment:
@privateip, this function seems incomplete... `INTF_NAME_RE` is never defined, nor are `first_port` or `last_port` below. Sorry I didn't catch it earlier on the PR, but now that everything is finally together, this might be an issue.